pkgs/build-support/setup-hooks/strip.sh

```diff
@@ -4,33 +4,77 @@ fixupOutputHooks+=(_doStrip)
 
 _doStrip() {
     if [ -z "$dontStrip" ]; then
-        stripDebugList=${stripDebugList:-lib lib32 lib64 libexec bin sbin}
-        if [ -n "$stripDebugList" ]; then
-            stripDirs "$stripDebugList" "${stripDebugFlags:--S}"
+        if [ -n "${stripDebugList:-*}" ]; then
+            stripDirs "${stripDebugList:-*}" "${stripDebugFlags:--S}"
         fi
 
-        stripAllList=${stripAllList:-}
-        if [ -n "$stripAllList" ]; then
-            stripDirs "$stripAllList" "${stripAllFlags:--s}"
+        if [ -n "${stripAllList:-}" ]; then
+            stripDirs "${stripAllList:-}" "${stripAllFlags:--s}"
         fi
     fi
 }
```

Comment:
What does an empty string mean ? Strip everything or nothing ?
For now, passing an empty string to stripDirs strips nothing because it is interpreted as a gob pattern. This is why "" is replaced by "*".

If you want to disable stripping, why not use dontStrip ?

My fear is that if the variable is defined for some reason, or remains (say, from another output), the result may not be expected by the user.

I may however remove the useless `if`. Would that suit you ?